zallet/src/components/json_rpc/methods/view_transaction.rs

```diff
@@ -22,28 +33,94 @@ use crate::components::{
     },
 };
 
+const POOL_TRANSPARENT: &str = "transparent";
+const POOL_SAPLING: &str = "sapling";
+const POOL_ORCHARD: &str = "orchard";
+
+/// The number of blocks within expiry height when a tx is considered to be expiring soon.
+const TX_EXPIRING_SOON_THRESHOLD: u32 = 3;
+
 /// Response to a `z_viewtransaction` RPC request.
 pub(crate) type Response = RpcResult<ResultType>;
 pub(crate) type ResultType = Transaction;
 
-/// Detailed shielded information about an in-wallet transaction.
+/// Detailed information about an in-wallet transaction.
 #[derive(Clone, Debug, Serialize, Documented, JsonSchema)]
 pub(crate) struct Transaction {
     /// The transaction ID.
     txid: String,
 
+    /// The transaction status.
+    ///
+    /// One of 'mined', 'waiting', 'expiringsoon' or 'expired'.
+    status: &'static str,
+
+    /// The number of confirmations.
+    ///
+    /// - A positive value is the number of blocks that have been mined including the
+    ///   transaction in the chain. For example, 1 confirmation means the transaction is
+    ///   in the block currently at the chain tip.
+    /// - 0 means the transaction is in the mempool. If `asOfHeight` was set, this case
+    ///   will not occur.
+    /// - -1 means the transaction cannot be mined.
+    confirmations: i32,
+
+    /// The hash of the main chain block that this transaction is mined in.
+    ///
+    /// Omitted if this transaction is not mined within a block in the current best chain.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    blockhash: Option<String>,
+
+    /// The index of the transaction within its block's `vtx` field.
+    ///
+    /// Omitted if this transaction is not mined within a block in the current best chain.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    blockindex: Option<u16>,
```

Comment:
Is there a reason to use u16 here? I ask because the transparent `OutPoint` type uses a u32; I ran into this in the `z_listunspent` transparent additions, where to use a u16 I would have to (fallibly) convert to the narrower representation; since this struct is just used for serialization, why not use a `u32` and avoid that conversion?

Reply:
We have consensus checks that Orchard bundles can have at most `2^16` actions, and the consensus transaction size limits have much lower limits for all the protocols, so I've tended to default to `u16` in all new places. But yeah, I guess as long as this is only used for serialization (which should be the case; we are never going to provide client libraries based on our code) then these relaxations should be fine.